Packs/Base/Scripts/CommonServerPython/CommonServerPython.py

```diff
@@ -10879,6 +10882,31 @@ def xsiam_api_call_with_retries(
     return response
 
 
+def split_data_to_chunks(data, chunk_size):
+    """
+    Splits a string of data into chunks of a specified size.
+
+    :type data: ``str``
+    :param data: The string of data to split.
+    :type chunk_size: ``int``
+    :param chunk_size: The maximum size of each chunk.
+
+    :return: A list of chunks containing the data.
+    :rtype: ``list``
+    """
+    chunk = []
+    curr_size = 0
+    for data_part in data.split('\n'):
```

Comment:
Shouldn't this function also support data that's a `list`?

Reply:
in  send_events_to_xsiam flow
If data is a list it is joined into a str.
So data is a string.

I can however not do the join to a string if data is a list.
But then if data is a string will I split it to a list ?

Or should I check in the split_data_to_chunks method if data is a list or a string ? 
I prefer defined data structures. at least in terms of collections vs. str or int 
But I understand it might not be a valid preference in python.
What do you say ?

Reply:
Python is very tolerable for both. In this case, it'll make much more sense for this function to get a `list` as an input rather than an `str` with a `'\n'` delimiter - so the default case should be a `list` rather an `str` - if you prefer to support just one usecase, I suggest to support a `list`.
Of course it'll be easy to support both usecases as the `for` loop iterates a list - making it support both just require an extra if before the `for` loop.